src/modeler/qubot.rs

```diff
@@ -360,6 +361,76 @@ impl<'a> Qubot<'a> {
                 self.qubo.add_rule(&new_qubit, Rule::Invalid);
                 self.record_mapping(gate, new_qubit)
             }
+            Gate::Quotient { index, .. } => {
+                let new_qubit = QubitRef::from(RefCell::new(Qubit {
+                    name: self.get_current_index(),
+                }));
+
+                let gate_key = HashableGateRef::from(gate.clone());
+                let nodes = self
+                    .bitblasting
+                    .constraint_based_dependencies
+                    .get(&gate_key)
+                    .unwrap();
+                let mut dividend: Vec<QubitRef> = Vec::new();
+
+                let mut node_key = HashableNodeRef::from(nodes.0.clone());
+                let mut temp_gates = self.bitblasting.mapping.get(&node_key).unwrap();
+                for t_gate in temp_gates {
```

Comment:
This could be written more succinctly as:
`let dividend = temp_gates.iter().map(|g| self.visit(g)).collect();`
Note sure this is easier to read, just wanted to mention it here as a suggestion.